assets/js/aplus.js

```diff
@@ -665,28 +669,22 @@ $(function() {
  * to the next
  * Note! Translation requires that at least 'hl' query parameter is copied
  */
-(function(window) {
-    window.addEventListener('click', function(event) {
-        if (
-            event.target.nodeName === 'A'
-            && event.target.href
-            && event.target.protocol === window.location.protocol
-            && event.target.host === window.location.host // hostname:port
-            && (event.target.search === '' || event.target.search === '?') // no search
-        ) {
-            event.preventDefault(); // stop default redirect
-            event.target.search = window.location.search; // copy query params
-            if (event.target.getAttribute('target') === '_blank') {
-                // Open a new window or tab.
-                window.open(
-                    event.target.href,
-                    '_blank',
-                    'location,menubar,resizable,scrollbars,status,noopener');
-            } else {
-                window.location.href = event.target.href; // redirect the current window
-            }
-        }
-    }, false);
+(function (window) {
+  // The following regex takes the language values from the LANGUAGES variable in the aplus/settings.py, i.e., en and fi.
+  // Therefore, the values must be adjusted accordingly.
+  const regexp = /\?+.*hl=(en|fi)+/gi;
+  const addQueryParameter = (event) => {
+    if (
+      event.target.protocol === window.location.protocol &&
+      event.target.host === window.location.host && // hostname:port
+      window.location.search.match(regexp) // hl search
+    ) {
+      event.target.search = window.location.search; // copy query params
```

Comment:
It might be a better idea to copy only the `hl` parameter here, instead of replacing all query parameters. [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) might help with that (note, no IE 11 support, but there is a polyfill of course).